src/compiler/moduleNameResolver.ts

```diff
@@ -1,4 +1,8 @@
 namespace ts {
+
+    const _fs: typeof import("fs") = require("fs");
+    const env_resolution_platforms = process.env['RESOLUTION_PLATFORMS'] && JSON.parse(process.env['RESOLUTION_PLATFORMS']);

```

Comment:
There's probably a tidier place for this, but I wanted to avoid re-inspecting the environment on every file access.  This may or may not matter (I don't know what caching `process.env` does).